data/src/test/java/io/gnosis/data/repositories/SafeRepositoryTest.kt

```diff
@@ -0,0 +1,207 @@
+package io.gnosis.data.repositories
+
+import android.app.Application
+import io.gnosis.data.db.daos.SafeDao
+import io.gnosis.data.models.Safe
+import io.mockk.*
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import pm.gnosis.ethereum.*
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.PreferencesManager
+import pm.gnosis.tests.utils.TestPreferences
+import pm.gnosis.utils.asEthereumAddressString
+import java.math.BigInteger
+
+class SafeRepositoryTest {
+
+    private val safeDao = mockk<SafeDao>()
+    private val ethereumRepository = mockk<EthereumRepository>()
+
+    private lateinit var preferences: TestPreferences
+    private lateinit var safeRepository: SafeRepository
+
+    @Before
+    fun setup() {
+        preferences = spyk()
+        val application = mockk<Application>().apply {
+            every { getSharedPreferences(any(), any()) } returns preferences
+        }
+        val preferencesManager = PreferencesManager(application)
+        safeRepository = SafeRepository(safeDao, preferencesManager, ethereumRepository)
+    }
+
+    @Test
+    fun `getSafes - should return safes in the database`() = runBlocking {
```

Comment:
should we use test names without spaces here?

Reply:
you mean like in instrumentation tests?

Reply:
I was thinking about having the same naming schema for all tests, but having spaces improves readability a lot. Let's use the same structure but spaces for unit tests and underscores for instrumentation.